src/pwm/pwm.c

```diff
@@ -61,7 +61,10 @@ mraa_pwm_write_period(mraa_pwm_context dev, int period)
         mraa_result_t result = dev->advance_func->pwm_period_replace(dev, period);
         if (result == MRAA_SUCCESS) {
             dev->period = period;
+        } else if (result == MRAA_ERROR_FEATURE_NOT_IMPLEMENTED) {
+            return MRAA_SUCCESS;
```

Comment:
I do not like the idea of doing it this way here, ie returning MRAA_SUCCESS for MRAA_ERROR_FEATURE_NOT_IMPLEMENTED. The other way would be to handle this in the UPM driver, but not sure if that is a good way to go about it either.